### PR TITLE
SONARPY-1915 implement S6929 for PyTorch

### PR DESCRIPTION
**Pull request by Seppli11**

### CODE REVIEW

**Comment by Seppli11** [comment]
How can I force the symbol to not be null, but not have a fully qualified name?

**Comment by guillaume-dequenne-sonarsource** [comment]
> How can I force the symbol to not be null, but not have a fully qualified name?

I think you can trigger this case if you import the same name twice from different places:
```
from something import foo
from something_else import foo

foo() # FQN: null
```

In this case, because the FQN of the symbol that is actually used can either be `something.foo` or `something_else.foo`, the FQN is set to null to avoid possible FPs.

File: python-checks/src/main/java/org/sonar/python/checks/TfPyTorchSpecifyReductionAxisCheck.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        createTfQuickFix(callExpression).ifPresent(issue::addQuickFix);
+      }
+
+      if (isPyTorchReductionMissingDimArg(symbol, callExpression)) {
+        context.addIssue(callExpression.callee(), PY_TORCH_MESSAGE);
+      }
+    }
+  }
+
+  private static boolean hasUnpackArgument(List<Argument> arguments) {
```
Comment (joke1196) [inline python-checks/src/main/java/org/sonar/python/checks/TfPyTorchSpecifyReductionAxisCheck.java:101]: I was not sure but did you already migrated this method to TreeUtils. I think I saw this one several times now.

File: python-checks/src/test/java/org/sonar/python/checks/TfPyTorchSpecifyReductionAxisCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    PythonQuickFixVerifier.verifyNoQuickFixes(new TfPyTorchSpecifyReductionAxisCheck(),
+        """
+        from tensorflow import math
+        math.reduce_all()
+        """);
+  }
+
+  @Test
+  void testPyTorch() {
+    PythonCheckVerifier.verify("src/test/resources/checks/pyTorchSpecifyReductionDim.py", new TfPyTorchSpecifyReductionAxisCheck());
```
Comment (joke1196) [inline python-checks/src/test/java/org/sonar/python/checks/TfPyTorchSpecifyReductionAxisCheckTest.java:63]: Here you could create a static variable with the `new TfPyTorchSpecifyReductionAxis()` that you can pass to the verifier.

**Review by joke1196 — APPROVED** [review]
Looks good! I have added a small comment but I will approve and you can merge once you made the change.

**Pull request merged**